server/app/models/grid_service.rb

```diff
@@ -179,7 +179,7 @@ def link_to(service, service_alias = nil)
   # @return [Hash]
   def env_hash
     if @env_hash.nil?
-      @env_hash = self.env.inject({}){|h, n| h[n.split('=', 2)[0]] = n.split('=', 2)[1]; h }
+      @env_hash = Hash[*self.env.flat_map { |kv_pair| kv_pair.split(/(?<!\\)=/, 2) }]
```

Comment:
This still fails with HTTP 500 errors on invalid envs...

```
ArgumentError (odd number of arguments for Hash):
  /home/kontena/kontena/kontena/server/app/models/grid_service.rb:182:in `[]'
  /home/kontena/kontena/kontena/server/app/models/grid_service.rb:182:in `env_hash'
  /home/kontena/kontena/kontena/server/app/mutations/grid_services/update.rb:120:in `build_grid_service_envs'
  /home/kontena/kontena/kontena/server/app/mutations/grid_services/update.rb:73:in `execute'
  /home/kontena/kontena/kontena/server/vendor/bundle/ruby/2.3.0/gems/mutations-0.7.2/lib/mutations/command.rb:80:in `run'
  /home/kontena/kontena/kontena/server/vendor/bundle/ruby/2.3.0/gems/mutations-0.7.2/lib/mutations/command.rb:32:in `run'
  /home/kontena/kontena/kontena/server/app/mutations/stacks/deploy.rb:47:in `block in create_or_update_services'
  /home/kontena/kontena/kontena/server/app/mutations/stacks/deploy.rb:43:in `each'
  /home/kontena/kontena/kontena/server/app/mutations/stacks/deploy.rb:43:in `create_or_update_services'
  /home/kontena/kontena/kontena/server/app/mutations/stacks/deploy.rb:31:in `execute'
  /home/kontena/kontena/kontena/server/vendor/bundle/ruby/2.3.0/gems/mutations-0.7.2/lib/mutations/command.rb:80:in `run'
  /home/kontena/kontena/kontena/server/vendor/bundle/ruby/2.3.0/gems/mutations-0.7.2/lib/mutations/command.rb:32:in `run'
  /home/kontena/kontena/kontena/server/app/routes/v1/stacks_api.rb:52:in `deploy_stack'
```

Would it be better to just `Hash[self.env.map{|env| env.split('=', 2) }]`? That seems to kinda work better (results in `nil` values for envs without `=`):

```ruby
Hash[['FOO'].map { |env| env.split('=', 2) }] => {"FOO"=>nil}
```